### PR TITLE
Added equals & hashcode method for PropertyEntry & it's subclasses.

### PR DESCRIPTION
**Pull request by dipak-pawar**
This change is [Reviewable](https://reviewable.io/reviews/arquillian/arquillian-recorder/51)

### CODE REVIEW

File: arquillian-recorder-reporter/arquillian-recorder-reporter-api/src/main/java/org/arquillian/recorder/reporter/model/entry/FileEntry.java [hunk 1]
```diff
@@ -86,4 +86,29 @@ public String getMessage() {
     public void setMessage(String message) {
         this.message = message;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
```
Comment (bartoszmajsak) [inline arquillian-recorder-reporter/arquillian-recorder-reporter-api/src/main/java/org/arquillian/recorder/reporter/model/entry/FileEntry.java]: I would consider changing this check to `instanceof` instead. It works for `null`s and can handle proxied classes.

**Comment by lordofthejars** [comment]
Looks good to me.

File: arquillian-recorder-reporter/arquillian-recorder-reporter-api/src/main/java/org/arquillian/recorder/reporter/model/entry/FileEntry.java (hunk 1 above)
Comment (lordofthejars) [inline arquillian-recorder-reporter/arquillian-recorder-reporter-api/src/main/java/org/arquillian/recorder/reporter/model/entry/FileEntry.java]: It looks good to me

Comment (dipak-pawar) [inline arquillian-recorder-reporter/arquillian-recorder-reporter-api/src/main/java/org/arquillian/recorder/reporter/model/entry/FileEntry.java]: Yes @lordofthejars.

@bartoszmajsak , instanceof check returns true for subclasses too, so its not strictly equals comparison

Comment (bartoszmajsak) [inline arquillian-recorder-reporter/arquillian-recorder-reporter-api/src/main/java/org/arquillian/recorder/reporter/model/entry/FileEntry.java]: @dipak-pawar got me :) Then we break the symmetry.

**Pull request merged**